BibBuddy/app/src/main/java/de/bibbuddy/LibraryFragment.java

```diff
@@ -224,40 +223,100 @@ private void handleDeleteShelf() {
     if (selectedShelfItems.size() > 1) {
       alertDeleteShelf.setTitle(R.string.delete_shelves);
       alertDeleteShelf.setMessage(
-          getString(R.string.delete_shelves_message) + " " + getString(R.string.delete_warning));
+          getString(R.string.delete_shelves_message)
+              + convertShelfListToString(selectedShelfItems)
+              + getString(R.string.delete_counter_msg)
+              + getBooksToDeleteNumber(selectedShelfItems)
+              + " und " + getNotesToDeleteNumber(selectedShelfItems) + " "
+              + getString(R.string.finally_delete) + " "
+              + getString(R.string.delete_warning));
     } else {
       alertDeleteShelf.setTitle(R.string.delete_shelf);
       alertDeleteShelf.setMessage(
-          getString(R.string.delete_shelf_message) + " " + getString(R.string.delete_warning));
+          getString(R.string.delete_shelf_message)
+              + convertShelfListToString(selectedShelfItems)
+              + getString(R.string.delete_counter_msg)
+              + getBooksToDeleteNumber(selectedShelfItems)
+              + " und " + getNotesToDeleteNumber(selectedShelfItems) + " "
+              + getString(R.string.finally_delete) + " "
+              + getString(R.string.delete_warning));
```

Comment:
Maybe you could write a separate function for the rest of the message because above you use the same strings. Only the first string part is different.